File: backend/src/main/java/com/beeswithmotion/thehive/post/Post.java
package com.beeswithmotion.thehive.post;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "posts")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long postId;

    private Long threadId;
    private String postContent;
    private String postAuthor;
    private LocalDateTime postDate;

    public Post() {}

    public Long getPostId() {
        return postId;
    }

    public Long getThreadId() {
        return threadId;
    }

    public String getPostContent() {
        return postContent;
    }

    public String getPostAuthor() {
        return postAuthor;
    }

    public LocalDateTime getPostDate() {
        return postDate;
    }

    public void setThreadId(Long threadId) {
        this.threadId = threadId;
    }

    public void setPostContent(String postContent) {
        this.postContent = postContent;
    }

    public void setPostAuthor(String postAuthor) {
        this.postAuthor = postAuthor;
    }

    public void setPostDate(LocalDateTime postDate) {
        this.postDate = postDate;
    }
}

File: frontend/src/types/post.ts
export type Post = {
    postId: number,
    threadId: number,
    postContent: string,
    postAuthor: string,
    postDate: string,
    image: ImageResponse | null
}

export type ImageResponse = {
    imageId: number,
    postId: number,
    originalFileName: string,
    contentType: string,
    uploadDateTime: string,
    imageUrl: string
}

export type CreatePostRequest = {
    postContent: string
}